pkg/image/api/helper.go

```diff
@@ -218,6 +218,20 @@ func SplitImageStreamTag(nameAndTag string) (name string, tag string, ok bool) {
 	return name, tag, len(parts) == 2
 }
 
+// SplitImageStreamImage turns the name of an ImageStreamImage into Name and Image ID.
+// It returns false if the tag was not properly specified in the name.
+func SplitImageStreamImage(nameAndImageID string) (name string, imageID string, ok bool) {
+	parts := strings.SplitN(nameAndImageID, "@", 2)
+	name = parts[0]
+	if len(parts) > 1 {
+		imageID = parts[1]
+	}
+	if len(imageID) == 0 {
+		imageID = ""
```

Comment:
if len(imageID)==0 then imageID already == "" so there's no point in setting it.